xoa/data_source_hosts.go

```diff
@@ -67,11 +71,18 @@ func dataSourceHostsRead(d *schema.ResourceData, m interface{}) error {
 func hostsToMapList(hosts []client.Host) []map[string]interface{} {
 	result := make([]map[string]interface{}, 0, len(hosts))
 	for _, host := range hosts {
+		cpus := map[string]string{
+			"sockets": fmt.Sprintf("%d", host.Cpus.Sockets),
+			"cores":   fmt.Sprintf("%d", host.Cpus.Cores),
+		}
 		hostMap := map[string]interface{}{
-			"id":         host.Id,
-			"name_label": host.NameLabel,
-			"pool_id":    host.Pool,
-			"tags":       host.Tags,
+			"id":           host.Id,
+			"name_label":   host.NameLabel,
+			"pool_id":      host.Pool,
+			"tags":         host.Tags,
+			"memory":       host.Memory.Size,
+			"memory_usage": host.Memory.Usage,
+			"cpus":         cpus,
```

Comment:
We should be able to replace this with `hostCpuInfoToMapList` as well. The types need to be updated, but then we don't need to duplicate this cpu string building code.

The following diff shows what I mean
```
diff --git a/xoa/data_source_host.go b/xoa/data_source_host.go
index 75c9d4e..f360ce3 100644
--- a/xoa/data_source_host.go
+++ b/xoa/data_source_host.go
@@ -78,10 +78,9 @@ func dataSourceHostRead(d *schema.ResourceData, m interface{}) error {
        return nil
 }

-func hostCpuInfoToMapList(host client.Host) map[string]interface{} {
-       cpus := map[string]interface{}{
+func hostCpuInfoToMapList(host client.Host) map[string]string {
+       return map[string]string{
                "sockets": fmt.Sprintf("%d", host.Cpus.Sockets),
                "cores":   fmt.Sprintf("%d", host.Cpus.Cores),
        }
-       return cpus
 }
diff --git a/xoa/data_source_hosts.go b/xoa/data_source_hosts.go
index 9d50bce..f228017 100644
--- a/xoa/data_source_hosts.go
+++ b/xoa/data_source_hosts.go
@@ -1,7 +1,6 @@
 package xoa

 import (
-       "fmt"
        "log"

        "github.com/ddelnano/terraform-provider-xenorchestra/client"
@@ -71,10 +70,6 @@ func dataSourceHostsRead(d *schema.ResourceData, m interface{}) error {
 func hostsToMapList(hosts []client.Host) []map[string]interface{} {
        result := make([]map[string]interface{}, 0, len(hosts))
        for _, host := range hosts {
-               cpus := map[string]string{
-                       "sockets": fmt.Sprintf("%d", host.Cpus.Sockets),
-                       "cores":   fmt.Sprintf("%d", host.Cpus.Cores),
-               }
                hostMap := map[string]interface{}{
                        "id":           host.Id,
                        "name_label":   host.NameLabel,
@@ -82,7 +77,7 @@ func hostsToMapList(hosts []client.Host) []map[string]interface{} {
                        "tags":         host.Tags,
                        "memory":       host.Memory.Size,
...skipping...
 }

-func hostCpuInfoToMapList(host client.Host) map[string]interface{} {
-       cpus := map[string]interface{}{
+func hostCpuInfoToMapList(host client.Host) map[string]string {
+       return map[string]string{
                "sockets": fmt.Sprintf("%d", host.Cpus.Sockets),
                "cores":   fmt.Sprintf("%d", host.Cpus.Cores),
        }
-       return cpus
 }
diff --git a/xoa/data_source_hosts.go b/xoa/data_source_hosts.go
index 9d50bce..f228017 100644
--- a/xoa/data_source_hosts.go
+++ b/xoa/data_source_hosts.go
@@ -1,7 +1,6 @@
 package xoa

 import (
-       "fmt"
        "log"

        "github.com/ddelnano/terraform-provider-xenorchestra/client"
@@ -71,10 +70,6 @@ func dataSourceHostsRead(d *schema.ResourceData, m interface{}) error {
 func hostsToMapList(hosts []client.Host) []map[string]interface{} {
        result := make([]map[string]interface{}, 0, len(hosts))
        for _, host := range hosts {
-               cpus := map[string]string{
-                       "sockets": fmt.Sprintf("%d", host.Cpus.Sockets),
-                       "cores":   fmt.Sprintf("%d", host.Cpus.Cores),
-               }
                hostMap := map[string]interface{}{
                        "id":           host.Id,
                        "name_label":   host.NameLabel,
@@ -82,7 +77,7 @@ func hostsToMapList(hosts []client.Host) []map[string]interface{} {
                        "tags":         host.Tags,
                        "memory":       host.Memory.Size,
                        "memory_usage": host.Memory.Usage,
-                       "cpus":         cpus,
+                       "cpus":         hostCpuInfoToMapList(host),
                }
                result = append(result, hostMap)
        }
```